wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -93,13 +93,15 @@ def load_path(
 
         extra_args = {}
         if version:
-            obj = self._s3.ObjectVersion(bucket, key, version).Object()
+            obj_version = self._s3.ObjectVersion(bucket, key, version)
             extra_args["VersionId"] = version
+            etag = self._etag_from_obj_version(obj_version)
+            obj = obj_version.Object()
         else:
             obj = self._s3.Object(bucket, key)
 
         try:
-            etag = self._etag_from_obj(obj)
+            etag = etag if version else self._etag_from_obj(obj)
```

Comment:
nit: lets ust grab the etag in the `if/else` branch so the code is easier to understand.